agent/consul/stream/topic_key.go

```diff
@@ -0,0 +1,36 @@
+package stream
+
+import "strings"
+
+// TopicKey identifies a portion of a topic for which a subscriber wishes to
+// receive events (e.g. health events for a particular service). It is used as
+// a map key when storing/retrieving event buffers and cached snapshots.
+//
+// Note: its fields are unexported as it's expected that users will call
+// NewTopicKey rather than constructing a TopicKey manually, as this method
+// normalizes the given values.
+type TopicKey struct {
+	key       string
+	namespace string
+	partition string
+}
```

Comment:
I think we might be better off with an opaque `TopicKey` similar to `Topic`. Nothing in the `stream` package should need to know about these individual fields. We could do something like:

```go
type TopicKey string
```

or similar to topic

```go
type TopicKey fmt.Stringer
```

Reply:
Makes sense! It probably makes more sense to do the normalization in `agent/rpc/subscribe` anyway 👍🏻